src/App.tsx

```diff
@@ -1,223 +1,84 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext, useEffect, useState,
+} from 'react';
+import { getTodos } from './api/todos';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodosFilter } from './components/TodosFilter';
+import { Header } from './components/Header';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { ErrorType } from './types/ErrorType';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todosFromServer, setUsersFromServer] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.NONE);
+  const [isError, setIsError] = useState(false);
+
+  const onFilter = useCallback(
+    (filteredTodos: Todo[]) => {
+      setVisibleTodos(filteredTodos);
+    }, [visibleTodos],
+  );
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
-    }
+    getTodos((user as User).id)
+      .then(result => {
+        setUsersFromServer(result);
+        setVisibleTodos(result);
+      })
+      .catch(() => {
+        setErrorType(ErrorType.LOAD);
+        setIsError(true);
+      });
   }, []);
 
+  const hideError = useCallback(
+    () => {
+      setIsError(false);
+      setTimeout(() => {
+        setErrorType(ErrorType.NONE);
+      }, 1500);
+    }, [],
+  );
+
+  useEffect(() => {
+    if (errorType !== ErrorType.NONE) {
+      setTimeout(() => {
+        hideError();
+      }, 3000);
+    }
+  }, [errorType]);
+
+  const isTodos = todosFromServer.length > 0;
```

Comment:
looks clearer this way
```suggestion
  const areTodosEmpty = !todosFromServer.length;
```